subsys/net/lib/azure_iot_hub/src/azure_iot_hub_cloud_api.c

```diff
@@ -30,6 +30,12 @@ static void api_event_handler(struct azure_iot_hub_evt *evt)
 				   config->user_data);
 
 		break;
+	case AZURE_IOT_HUB_EVT_CONNECTION_FAILED:
+		cloud_evt.type = CLOUD_EVT_ERROR;
```

Comment:
Should we add a ```CLOUD_EVT_CONNECTION_FAILED``` event and do a apples to apples conversion here, not just translating it to the generic ```CLOUD_EVT_ERROR``` event.

Reply:
Yeah we could do that in another PR for all backends